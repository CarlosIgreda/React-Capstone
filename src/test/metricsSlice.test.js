import metricsReducer, { asset } from '../redux/metrics/metricsSlice';

describe('metricsSlice reducer', () => {
  it('should handle asset', () => {
    const initialState = { data: null };
    const payload = { value: 10 };

    const newState = metricsReducer(initialState, asset(payload));

    expect(newState).toMatchSnapshot();
  });
});

describe('metricsSlice actions', () => {
  it('should create an action to set asset data', () => {
    const payload = { value: 10 };
    const expectedAction = {
      type: 'metrics/asset',
      payload,
    };

    expect(asset(payload)).toEqual(expectedAction);
  });
});
