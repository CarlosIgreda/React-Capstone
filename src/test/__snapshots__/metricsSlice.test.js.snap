// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`metricsSlice reducer should handle asset 1`] = `
Object {
  "data": Object {
    "value": 10,
  },
}
`;
